backend/workflow_manager/endpoint/destination.py

```diff
@@ -123,12 +131,27 @@ def validate(self) -> None:
         ):
             raise DestinationConnectorNotConfigured()
 
+    def _push_data_to_queue(
```

Comment:
NIT: I think we should have Endpoint as a baseclass and let other types inherit it so that we have functions grouped to that particular type. I guess it can only be handled in an eventual refactor. This will help avoid our logic of check the `connection_type`

Reply:
@chandrasekharan-zipstack _push_data_to_queue is only for destination. If something else needs to be refactored can do later in refactoring.